lua/treesitter-context.lua

```diff
@@ -536,13 +540,31 @@ local function open(ctx_nodes)
   highlight_contexts(bufnr, ctx_bufnr, contexts)
 end
 
+--- Get relative line between 2 line with folding calculation
+---@param line1 number The first line
+---@param line2 number The second line
+---@return number relativeline Relative line
+local function calc_relativeline(line1, line2)
+  local line, higher_line = line1, line2
+  if line1 > line2 then
+    line, higher_line = line2, line1
+  end
+  local relativeline = 0
+  local i = line
+  while i < higher_line do
+    i = get_next_line(i)
+    relativeline = relativeline + 1
+  end
+  return relativeline
+end
+
 local function calc_max_lines(config_max)
```

Comment:
@lewis6991, Do you think we should remove this function? I just notice that I have added more condition to make sure context not overflow cursor without calc max line.